pkg/collector/corechecks/embed/jmx.go

```diff
@@ -265,14 +272,24 @@ func (c *JMXCheck) start() error {
 
 	subprocessArgs := []string{}
 
-	// Specify a maximum memory allocation pool for the JVM
 	javaOptions := c.javaOptions
-	if !strings.Contains(javaOptions, "Xmx") && !strings.Contains(javaOptions, "XX:MaxHeapSize") {
-		javaOptions += jvmDefaultMaxMemoryAllocation
-	}
-	// Specify the initial memory allocation pool for the JVM
-	if !strings.Contains(javaOptions, "Xms") && !strings.Contains(javaOptions, "XX:InitialHeapSize") {
-		javaOptions += jvmDefaultInitialMemoryAllocation
+	if config.Datadog.GetBool("jmx_use_cgroup_memory_limit") {
+		// This option is incompatible with the Xmx and Xms options, log a warning if there are found in the javaOptions
+		for _, option := range jvmCgroupMemoryIncompatOptions {
+			if strings.Contains(javaOptions, option) {
+				log.Warnf("Java option %q is incompatible with cgroup memory limit, please remove it", option)
```

Comment:
Could we please explicitly state that the option it will be ignored.

Reply:
reworked the wording, and not passing the option, instead of passing it and having java silently ignore it